### PR TITLE
Add marks nightly and weakly for torch tests

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Add marks `nightly` and `weakly` for tests.
- Mark sanity tests as `nightly`
- Split `test_functions.TestParametrized` to fast for precommit and long for nightly
- Time of torch precommit reduced from 60 to 40 mins
- Set `xfail` for sanity tests with `--mode train` in case of segment fault.
  Sporadic segment fault reproduced on torch>=2.0.0 on call `backward` function.

### Related tickets

119128

### CODE REVIEW

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Could you think about move these tests from pre-commit scope to the nightly scope?

File: tests/torch/test_sanity_sample.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def test_pretrained_model_eval(config, tmp_path, multiprocessing_distributed, ca
         args["--multiprocessing-distributed"] = True
 
     runner = Command(create_command_line(args, config["sample_type"]), env=ROOT_PYTHONPATH_ENV)
-    runner.run()
+    runner.retries_run()  # WA for 119128
```
Comment (vshampor) [inline tests/torch/test_sanity_sample.py:214]: This will make any kind of failure trigger a retry. Redesign to retry on segmentation faults specifically.

**Review by vshampor — CHANGES_REQUESTED** [review]

**Review by alexsu52 — COMMENTED** [review]
Cool staff! Could you provide CI build for nightly precommit tests?

**Review by alexsu52 — APPROVED** [review]
Thanks for the contribution!

**Review by vshampor — CHANGES_REQUESTED** [review]
So where's the retry functionality in case of a segfault now? Otherwise we are getting a nightly build which will be often red due to a known issue and no one will bother to make it green, leading to overlooked bugs in the backend-nightly scope.

**Comment by AlexanderDokuchaev** [comment]
> So where's the retry functionality in case of a segfault now? Otherwise we are getting a nightly build which will be often red due to a known issue and no one will bother to make it green, leading to overlooked bugs in the backend-nightly scope.

All changes exist in descript of PR. Yes, seg fault from original ticket will be moved to nightly job.

**Review by daniil-lyakhov — APPROVED** [review]

**Comment by vshampor** [comment]
Should be merged after #2112

**Comment by AlexanderDokuchaev** [comment]
> Should be merged after https://github.com/openvinotoolkit/nncf/pull/2112

why?

**Comment by vshampor** [comment]
@AlexanderDokuchaev see my [comment](https://github.com/openvinotoolkit/nncf/pull/2092#pullrequestreview-1608562245) above

**Comment by AlexanderDokuchaev** [comment]
> @AlexanderDokuchaev see my https://github.com/openvinotoolkit/nncf/pull/2092#pullrequestreview-1608562245 above

Your comment it's not the reason to block this PR.

**Comment by alexsu52** [comment]
@vshampor, let's merge it.

**Review by alexsu52 — APPROVED** [review]

**Comment by AlexanderDokuchaev** [comment]
run pylint pre-commit tests
run pytorch pre-commit tests
run openvino pre-commit tests
run onnx pre-commit tests

**Pull request merged**